crates/cast/src/starknet_commands/script/init.rs

```diff
@@ -0,0 +1,141 @@
+use clap::Args;
+use indoc::{formatdoc, indoc};
+use scarb_metadata::ScarbCommand;
+
+#[derive(Args, Debug)]
+pub struct Init {
+    /// Name of a script to create
+    pub script_name: String,
+}
+
+pub fn init(init_args: &Init) -> Result<ScriptInitResponse> {
+    let script_root_dir_path = get_script_root_dir_path(&init_args.script_name)?;
+
+    init_scarb_project(&init_args.script_name, &script_root_dir_path)?;
+    add_dependencies(&script_root_dir_path)?;
+    modify_files_in_src_dir(&init_args.script_name, &script_root_dir_path)?;
+
+    Ok(ScriptInitResponse {
+        status: format!("Successfully initialized `{}`", init_args.script_name),
+    })
+}
+
+fn get_script_root_dir_path(script_name: &str) -> Result<Utf8PathBuf> {
+    let current_dir = std::env::current_dir()?;
+
+    let script_root_dir_path = current_dir
+        .file_name()
+        .and_then(|dir_name| dir_name.to_str())
+        .filter(|&dir_name| dir_name == SCRIPTS_DIR)
+        .map_or_else(
+            || current_dir.join(SCRIPTS_DIR).join(script_name),
+            |_| current_dir.join(script_name),
+        );
+
+    Utf8PathBuf::from_path_buf(script_root_dir_path)
+        .map_err(|_| anyhow!("Failed to obtain script root dir path"))
+}
+
+fn init_scarb_project(script_name: &str, script_root_dir: &Utf8PathBuf) -> Result<()> {
+    ScarbCommand::new()
+        .args([
+            "new",
+            "--name",
+            &script_name,
+            "--no-vcs",
+            "--quiet",
+            script_root_dir.as_str(),
+        ])
+        .run()
+        .context("Failed to init Scarb project")?;
+
+    Ok(())
+}
+
+fn add_dependencies(script_root_dir: &Utf8PathBuf) -> Result<()> {
+    add_sncast_std_dependency(script_root_dir)
+        .context("Failed to add sncast_std dependency to Scarb.toml")?;
+    add_starknet_dependency(script_root_dir)
+        .context("Failed to add starknet dependency to Scarb.toml")?;
+
+    Ok(())
+}
+
+fn add_sncast_std_dependency(script_root_dir: &Utf8PathBuf) -> Result<()> {
+    let cast_version = env!("CARGO_PKG_VERSION");
+    let cast_version = format!("v{cast_version}");
+
+    ScarbCommand::new()
+        .current_dir(script_root_dir)
+        .args([
+            "--offline",
+            "add",
+            "sncast_std",
+            "--git",
+            "https://github.com/foundry-rs/starknet-foundry.git",
+            "--tag",
+            &cast_version,
+        ])
+        .run()?;
+
+    Ok(())
+}
+
+fn add_starknet_dependency(script_root_dir: &Utf8PathBuf) -> Result<()> {
+    let scarb_manifest_path = script_root_dir.join("Scarb.toml");
+    let cairo_version =
+        get_cairo_version(&scarb_manifest_path).context("Failed to obtain cairo version")?;
+    let starknet_dependency = format!("starknet@>={cairo_version}");
+
+    ScarbCommand::new()
+        .current_dir(script_root_dir)
+        .args(["--offline", "add", &starknet_dependency])
+        .run()?;
+
+    Ok(())
+}
```

Comment:
is there is there a reason not to combine those two functions?

Reply:
Same comment as [here](https://github.com/foundry-rs/starknet-foundry/pull/1298/files#r1417573712)

Reply:
I have mixed feelings, as right now we actually use 3 functions to just add 2 dependencies (add_starknet_dependency, add_sncast_std_dependency and add_dependencies). I'm not sure we need all three to be honest. I

Reply:
as per https://github.com/foundry-rs/starknet-foundry/pull/1298#discussion_r1417477652